### ISSUE TITLE
Splash Images are not displaying from the storyboad in 33.1.1.176 

### ISSUE THREAD

**Issue by ukmunnithan** [Gamua/Adobe-Runtime-Support#381]
## Problem Description
I have created the splash images from the storyboard (https://wiki.starling-framework.org/manual/ios_launch_images#using_launchscreenstoryboard) as like this way. And added the storyboard file and assets.car file in my project root directory. included the  tag <key>UILaunchStoryboardName</key>
    <string>LaunchScreen</string>in the app.xml file.

But when the app is launching it shows a black screen instead of splash images in the iPhones and a white screen in the iPad instead of the splash images. 
Could you please help

* Which AIR SDK did you use?  33.1.1.176
* Describe the affected environment: Flashbuilder 4.7. iOS (iPad and iPhone devices)

I am attaching storyboard file and assets.car files
[Archive.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4883439/Archive.zip)

**Comment by stoff99** [comment]
Hey, do you have a valide licence file in the sdk/lib folder?
After i buy the license and added the file i was able to see the storyboards.
Harman changed there "trial" image also to a storyboard. They are in conflict than i think. In this case i was not able to see my storyboard and also not the storyboard from Harman. ;)

**Comment by ukmunnithan** [comment]
@stoff99 Yes, I have a licence file and copied it in the lib folder.
The trial splash screen also not displaying while launching the app..
@ajwfrost  @marchbold  Please help

**Comment by ukmunnithan** [comment]
@stoff99 could you please use the assets.car file and LaunchScreen.storyboardc file included in the zip file and check whether its showing on your app. I just need to confirm whether the issue with these files or Harman sdk ..

Or could you please share the sample project that you have created for generating assets.car file and LaunchScreen.storyboardc? So that I can check it from my side itself.

Thanks in advance

**Comment by stoff99** [comment]
Hello @ukmunnithan,
i can try your assets car and storyboard.
I can just recomment this docu from Distriqt for Assets.car and simply launch image: https://airnativeextensions.github.io/tutorials/ios-icons-assets-car#method-3-online-tool

You create your simple XCode project and add the app icon. Than you can simply add a new image set and name it LaunchImage and place your launch image there. Use the dimension 2732x2732 and place your content in the middle of this big image.
Here is a screenshot how it should look like:
![Assets_xcassets](https://user-images.githubusercontent.com/892820/86779474-e8e37880-c05b-11ea-81f2-0e6e8e54dc3e.png)


Than download the premade storyboard from Distriqt: https://airnativeextensions.github.io/tutorials/resources/ios/LaunchScreen.storyboardc.zip
and add it to your root next to your assets.car which includes your app icon and your LaunchImage.
This storyboard will take the LaunchImage automatically out of you assets.car and place it in the center.

Dont forget the InfoAdditions entry:
<key>UILaunchStoryboardName</key>
<string>LaunchScreen</string>

Also make sure you have include the assets.car and storyboard file/folder in the packaging.

**Comment by ukmunnithan** [comment]
@stoff99 Thanks for your help. I will try and let you know the results.
I followed almost the same steps. Anyway I will try again

**Comment by stoff99** [comment]
@ukmunnithan ,
i added your assets.car and storyboard.
i can see your app icon + the launch screen. The launch screen logo is stretched but it will be displayed.
See the screenshot:
![IMG_0099](https://user-images.githubusercontent.com/892820/86780680-49bf8080-c05d-11ea-8148-87fbd867683c.PNG)

are you sure you placed the .lic file in the correct sdk folder /lib?
And you reference to this sdk in your project where you placed the .lic file?
![lib_und_LaunchImage_imageset_und_523526_und_plentyco_logo](https://user-images.githubusercontent.com/892820/86780838-796e8880-c05d-11ea-85cf-eeb73b101fc1.png)

**Comment by ukmunnithan** [comment]
@stoff99 Thank you for your help. I had placed the adt.lic file in the lib folder itself.
Do we need to take release build to see this? or let it be visible while debugging the app?

**Comment by ajwfrost** [comment]
Hi - in case it helps, when we were debugging storyboard issues, we generally extracted the IPA that is generated by the AIR SDK, and then checked (a) there is a .storyboardc folder; (b) there is a 'UILaunchStoryboardName' entry in the Info.plist file that refers to this storyboard name; (c) the assets.car file has the right images - case sensitive - as expected by the storyboard (use 'assetutil -I assets.car' for details).

thanks

**Comment by stoff99** [comment]
@ukmunnithan yes it will also be displayed in debug mode. You dont need to create an adhoc or release build for this.
I currently realized that the launch image will be not displayed on my iPhone 11 Pro. I think that the storyboard file, which is provided by distriqt is outdated.
I will now make my own storyboard with the other guide: https://airnativeextensions.github.io/tutorials/ios-launchscreens

**Comment by marchbold** [comment]
@stoff99 What makes you think the storyboard is outdated? It's more likely that the asset catalog is missing an asset version for the device than the storyboard is outdated.

**Comment by ukmunnithan** [comment]
@ajwfrost  I have extracted the iPA file and all the files are present there
 assets.car, .storyboardc files present and info.plist file contains the UILaunchStoryboardName tag
@stoff99 
And I have copied the adt.lic file in the SDK/lib folder.
But the white screen is showing while opening the app.

**Comment by stoff99** [comment]
@marchbold @ukmunnithan ,
the whole stuff is a bit strange on my side.
I replaced my storyboard and assets.car with the files frim @ukmunnithan and i saw the launch image on the iPhone 11 pro.
Than i reverted this changes and tried again with my storyboard and assets.car and than didn't saw the launch image on the iPhone 11 pro.
Than i added the launch image in the Assets.xcassets for the 2x and 3x and added the assets.car again and than it was visible again.
Than i removed the 2x and 3x again and rebuild the assets.car and than it was still visible on the 11 pro ...

Its a bit magic whats going on here ;)

**Comment by stoff99** [comment]
@ukmunnithan on which device are you testing?

**Comment by ukmunnithan** [comment]
@stoff99 I have checked in iPhoneXR and iPad 2 device.

**Comment by ukmunnithan** [comment]
@stoff99 Here is the sample project I have done. In this also its not showing. Could you please run this app and check whether its showing or not?
[SampleApp.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4889024/SampleApp.zip)

**Comment by stoff99** [comment]
Hello @ukmunnithan , sry but i have no active flex sdk and setup.
I only develop in pure AS3.

**Comment by ajwfrost** [comment]
@stoff99 @ukmunnithan  one thing I also noticed with storyboards is that iOS seems to cache some of the details somehow. When our developer was telling me it was working, and I was seeing just a white screen during start-up, I was getting somewhat frustrated with him .. but then after reading something on stackoverflow, I tried uninstalling the application, rebooting the phone, and reinstalling the application - and this time it then worked.

I wonder whether the assets for an application are held/cached centrally and not updated if an app is just updated, perhaps..?

Anyway - if you're seeing odd behaviour, it's definitely worth trying the above steps so that you're using a clean-ish environment each time!

thanks

**Comment by stoff99** [comment]
@ajwfrost good to hear that also other persons struggle with that. :)
Its working now again and i will not change it again for sure. :D
I also deinstalled the app fully but i didn't rebooted the phone.
Maybe its also an issue with the icloud which is creating app backups.

**Comment by ukmunnithan** [comment]
@ajwfrost I had done the steps (uninstalled, rebooted the device), But no changes, still getting white screen

**Comment by stoff99** [comment]
@ukmunnithan , i had a call with a native iOS developer. This often happens also for him in debug build mode.
Can you try to create an adhoc or an production build, upload to Apple and install with TestFlight?

**Comment by stoff99** [comment]
@ukmunnithan some news related to your problem? I'm curious :)

**Comment by ukmunnithan** [comment]
@stoff99 @ajwfrost  I have tried with new SDK 33.1.1.190 and the issue still persists with my app.
I could not able to see the splashscreen from the storyboard

**Comment by stoff99** [comment]
strange, did you tried with another storyboard and assets file?
should i send you my one? Just to be sure the problem isn't related to this 2 files? But on other hand, yours was working in my app so ...

**Comment by ukmunnithan** [comment]
@stoff99 Yes, that will be helpful. Please share your storyboard file and assets.car file, so that I will try using your file and let you know the result

**Comment by stoff99** [comment]
ok please send me your e-mail address. i will send you the files.

**Comment by ukmunnithan** [comment]
Thank you! my email id is muralee@fingent.com

**Comment by stoff99** [comment]
ok you should got it already. Let me know if something has changed with this 2 files.

**Comment by ukmunnithan** [comment]
@stoff99 I have tried with your files and its not displayed
@ajwfrost @stoff99 One doubt whether the splash screen from storyboard will reflect only for devices with os>13.
Do we need to proceed the old method for devices os <13?

I also removed the adt.lic file then also the Harman splashscreen is not displayed.

**Comment by stoff99** [comment]
Hey @ajwfrost , thats rly strange in your case.
What happens when you create a empty new project with your assets and storyboard?
About older icons and launch image: https://github.com/distriqt/AIR-ImageScripts
Icons are supported with iOS 11+ via Assets.car. If you support older version you need to include them.

**Comment by ukmunnithan** [comment]
I have created an empty project, then also result is same getting a white screen instead of splash image
I am using the icons from the Assets.car file and icon is displayed. The issue is only with splash image.
My query is related to splash image. Do we need to package different sized spalsh image in the root to support ios 10,11 and 12

**Comment by ajwfrost** [comment]
Hi - just an extra point here, if you're using your own icons in an Assets.car file, and are expecting to see the Harman splash screen, then you need to merge our resources with your ones (see release notes). My understanding though is that you've got a license file so this is your own storyboard and assets that are being used?

**Comment by ukmunnithan** [comment]
Yes, we have our own icon and storyboard file. Its not showing. To
troubleshoot it, I have removed our assets.car file and storyboard file
then also I removed the licence file too. Then Harman splashscreen is not
showing.

**Comment by ajwfrost** [comment]
Okay thanks .. we can try grabbing your SampleApp zip file above and trying with this .. 
You're using MacOS I can see, just to check, what version do you have?
thanks

**Comment by ukmunnithan** [comment]
Thanks. My os version is Mojave 10.14.6

**Comment by ajwfrost** [comment]
@ukmunnithan looking at your storyboard file, I’m not sure that it’s properly set up to use the “solash_landscape” etc file that’s in the Assets file. You’re using a different Xcode version from us which means you’re getting the packed Nib format but if I look in the nib file within the storyboard, there’s no mention of “splash_landscape”..

Is it possible for you to share the .storyboard file, or can you confirm what images you’re actually expecting to see? The Assets includes the splash_*.png files and also some launch images but I can’t see any references in the storyboard files for these, whereas when I look at the storyboard files that we generate here, we can see there’s some names used in them that refer back to the assets..

Thanks

**Comment by ukmunnithan** [comment]
@ajwfrost Hereby attaching the storyboard file and assets.car file.
[Archive.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4923108/Archive.zip)

**Comment by ajwfrost** [comment]
@ukmunnithan I was hoping for the storyboard source code rather than the compiled storyboardc folder, but actually if we do look into this:
LaunchScreen.storyboardc/01J.....nib
It doesn’t contain “splash” in there, but it does contain “LaunchIcon”. So presumably this is the image that you’re trying to display I.e. Default-414w-736h@3x~iphone.png

Looking further at the assets, in our file we have two entries for each image, one is compressed with deepmap-lzfse which is what yours uses, and the other uses lzvn. So this may be to do with how you’re embedding your images, or the project/compatibility settings within this perhaps?

**Comment by ukmunnithan** [comment]
Hi @ajwfrost 
We have modified by including the splash_* images. I am sharing the source code of the assets.car file and storyboard file.
Could you please check this and check whether its working or not?
[AssetImage.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4947853/AssetImage.zip)

**Comment by ajwfrost** [comment]
Hi
Just looking at this, it's set to build for "iPhone 11 Pro Max" and this is then limited the assets that are exported into the Assets.car file.. are you able to change this and build for "Generic iOS Device" and try with the Assets.car from that?
thanks

**Comment by ukmunnithan** [comment]
While taking build I have selected device as 'Generic ios device'.

**Comment by ukmunnithan** [comment]
@ajwfrost @marchbold @stoff99
Hi when uploading the build on testflight, We are getting the message as ERROR ITMS-90596: "Invalid Bundle. The asset catalog at 'Payload/xxxxxx.app/Assets.car' can't be processed. Rebuild your app, and all included extensions and frameworks, with the latest GM version of Xcode and resubmit."
I have created the assets.car and storyboard file using XCode 11.3 with deployment target 10.0.
Could you please help

**Comment by ajwfrost** [comment]
@ukmunnithan  okay just to go back to the storyboard/assets: if we create the assets.car and storyboardc using Xcode with generic iOS device, and then take those two into the IPA packaging on a licensed version of the SDK, along with an XML descriptor file with the UILaunchStoryboardName key, then it works as we would expect.. your assets file is used showing the right application icon, and the right storyboard launch image is shown.

There's an issue if you're using "-connect" to package up as the 'connect to debugger' dialog is shown below the storyboard which then never is removed!

In terms of the TestFlight issue above, this may the same as was just reported where our IPA files are advertised as being created by Xcode 11.2.0 which is no longer supported, we can update this flag so I'll get you a version to try out.. 

Have you got the storyboard stuff working too or do you have open issues on that? We're definitely finding that we need to uninstall the app from the device/simulator each time otherwise it doesn't clear out the settings..

thanks

**Comment by ukmunnithan** [comment]
@ajwfrost Thank you for kind support. 
Yes,  the launch icon and app icon are now showing from the assets.car and the storyboard.
So that i have taken the build to test in all the devices after uploading in testflight. Then we are getting this issue...

**Comment by ajwfrost** [comment]
@ukmunnithan  in case you didn't see it, we've got a test version of ADT available which can be put into the "lib" folder in the SDK, and will then update the Xcode version number in the Info.plist file. So this may be behind that error you're seeing about 'Invalid Bundle'..?
https://transfer.harman.com/message/Bu00u9yZUoYTyHCOc5fekm

**Comment by ukmunnithan** [comment]
@ajwfrost Thank you for sharing the file. 
I have created the assets.car file using xcode 11.3 so do i need to use the same adt for building  the IPA file. 
In the base sdk you have used 11.2.0 whether this will resolve if updated to 11.2.1? 
As per apple documentation they states use xcode version 11.not mentioned about subversko

**Comment by ukmunnithan** [comment]
@ajwfrost I am getting the same message when trying to upload the build-in testflight even after I used the new adt.jar file.
ERROR ITMS-90596: "Invalid Bundle. The asset catalog at 'Payload/xxxxx.app/Assets.car' can't be processed. Rebuild your app, and all included extensions and frameworks, with the latest GM version of Xcode and resubmit."

Do I need to create the ANE's in the latest XCode version?

**Comment by ukmunnithan** [comment]
@ajwfrost Previously We have created and uploaded build with 33.1.1.98.Now we are getting this issue. So I think the issue is not related to ane's. For 33.1.1.190 we are newly included the asset.car file and storybord file.. 
Now our app release is holding because of this issue. Please help

**Comment by ukmunnithan** [comment]
@ajwfrost Thanks for the support. 
My issue is resolved. I have uploaded the build to Testflight...

**Comment by ajwfrost** [comment]
Okay great! thanks - any hints as to what you had to do in order to get it accepted would perhaps be useful for others who encounter similar issues! :-) 
cheers

**Comment by ukmunnithan** [comment]
We were used the Application Loader to upload the build to Testflight.  And we were used the tool for last 4-5 years. 
Yesterday we have tried with Transporter app to upload the build and it was successful. 
@ajwfrost sorry for the trouble encountered for the last 2 days. Anyway Thanks for the support

**Comment by ukmunnithan** [comment]
@ajwfrost. My issue still persit related flash effect in the splashimages and I have posted one comment and attachment in #334

**Comment by ukmunnithan** [comment]
@ajwfrost I am experiencing a strange issue. I placed two splash screen images one with CBN Logo and one with Superbook Logo. Both images are made with the same background color. CBN logo is placed in the storyboard file and superbook logo I placed in the Flex splashscreenimage. 
When the app opens in the first time the image from storyboard shows in a brighter colour then on the subsequent openings the color keeps the same.
Please refer the attached image 
![Comparison](https://user-images.githubusercontent.com/13150122/89883759-3bc5c800-dbe6-11ea-8fe2-9890e27660cc.png)
